Dockerfile

```diff
@@ -28,10 +28,6 @@ USER airflow
 COPY requirements.txt /
 RUN pip install --no-cache-dir -r /requirements.txt
 
-# Override default version from the constraints file because it is broken
-# https://github.com/apache/airflow/pull/34194
-RUN pip install apache-airflow-providers-google==8.11.0 apache-airflow-providers-cncf-kubernetes==5.2.2 --upgrade
```

Comment:
@mikaeld it looks like [your fix](https://github.com/apache/airflow/pull/34194/files#diff-95ed7631b172975c14b35fb4eeefebef992f7a1d1631d47a7222587ed94d8a07L542-L545) from the referenced PR [isn't included in Airflow 2.7.3](https://github.com/apache/airflow/blob/2.7.3/airflow/providers/google/cloud/operators/kubernetes_engine.py#L541-L544), so I'd think this would still be a problem, despite my minimal testing not showing any issues.

Do you recall what DAGs/tasks were impacted by that problem?

Reply:
Your fix [is being included in Airflow 2.8.0](https://github.com/apache/airflow/blob/2.8.0b1/airflow/providers/google/cloud/operators/kubernetes_engine.py#L542-L546) so maybe we should wait for that?

Reply:
![image](https://github.com/mozilla/telemetry-airflow/assets/1668835/73b2dafd-1557-40e6-9b94-585659c0e7df)
Provider packages are independent from Airflow releases. The constraints file sets `apache-airflow-providers-google=10.11.0` and my fix [was included in 10.9.0](https://airflow.apache.org/docs/apache-airflow-providers-google/stable/commits.html#id5)

Reply:
[This is the right branch](https://github.com/apache/airflow/blob/7574e16e751e37cc012139da1a0e39874bab2918/airflow/providers/google/cloud/operators/kubernetes_engine.py#L543), and you can see the fixed code is included